--- a/src/http/modules/ngx_http_limit_req_module.c	2012-09-24 22:11:45.000000000 +0300
+++ b/src/http/modules/ngx_http_limit_req_module.c	2013-01-24 17:48:48.532793440 +0200
@@ -245,7 +245,7 @@
             ctx->node = NULL;
         }
 
-        return NGX_HTTP_SERVICE_UNAVAILABLE;
+        return NGX_HTTP_TOO_MANY_REQUESTS;
     }
 
     /* rc == NGX_AGAIN || rc == NGX_OK */
--- a/src/http/ngx_http_request.h	2012-07-02 20:41:52.000000000 +0300
+++ b/src/http/ngx_http_request.h	2013-01-24 17:49:33.204790386 +0200
@@ -91,6 +91,8 @@
 #define NGX_HTTP_UNSUPPORTED_MEDIA_TYPE    415
 #define NGX_HTTP_RANGE_NOT_SATISFIABLE     416
 
+/* Custom rate limiting response */
+#define NGX_HTTP_TOO_MANY_REQUESTS         429
 
 /* Our own HTTP codes */
 
